codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/RestJson.kt

```diff
@@ -47,6 +49,44 @@ class RestJsonFactory : ProtocolGeneratorFactory<HttpBoundProtocolGenerator> {
     }
 }
 
+/**
+ * This [HttpBindingResolver] mostly delegates to [HttpTraitHttpBindingResolver].
+ */
+class RestJsonHttpBindingResolver(
+    model: Model,
+    contentTypes: ProtocolContentTypes,
+) : HttpBindingResolver {
```

Comment:
Why do we need this new class delegating to `HttpBindingResolver`? I understand the idea behind setting the content-type in responses.. That said, shouldn't we do this for any protocol in an agnostic way?

Reply:
It is a custom detail of the protocol. Until now we have not needed a `RestJson.kt` class because the client behavior of the protocol can be entirely described by Smithy's vanilla HTTP binding traits, which is what `HttpTraitHttpBindingResolver` is in charge of. But for the server, default response `Content-Type` is a protocol-specific behavior that is not configurable with Smithy traits alone. In any case, it's best that we have a separate class for `RestJson1` like we do for the other protocols, even if the implementation is very simple and just delegates.

I've clarified the docs.